include/clang/Constraints/Constraints.h

```diff
@@ -0,0 +1,457 @@
+//                     The LLVM Compiler Infrastructure
```

Comment:
Constraints.cpp / Constraints.h are an interface that's more general than clang, so I made Constraints a library. I was envisioning a potential future where something else would use the Constraint interface. That future might never come, though. They could move into the folder that has the rest of the source code for the tool.